src/core/transaction.cpp

```diff
@@ -99,27 +99,36 @@ namespace mamba
         if (m_expected_size && size_t(m_target->downloaded_size) != m_expected_size)
         {
             LOG_ERROR << "File not valid: file size doesn't match expectation " << m_tarball_path
-                      << "\nExpected: " << m_expected_size << "\n";
+                      << "\nExpected: " << m_expected_size
+                      << "\nActual: " << size_t(m_target->downloaded_size) << "\n";
             m_progress_proxy.mark_as_completed("File size validation error.");
             m_validation_result = SIZE_ERROR;
+            return;
         }
         interruption_point();
 
-        if (!m_sha256.empty() && !validate::sha256(m_tarball_path, m_sha256))
+        if (!m_sha256.empty())
         {
-            m_validation_result = SHA256_ERROR;
-            m_progress_proxy.mark_as_completed("SHA256 sum validation error.");
-            LOG_ERROR << "File not valid: SHA256 sum doesn't match expectation " << m_tarball_path
-                      << "\nExpected: " << m_sha256 << "\n";
+            auto sha256sum = validate::sha256sum(m_tarball_path);
+            if (m_sha256 != sha256sum)
+            {
+                m_validation_result = SHA256_ERROR;
+                m_progress_proxy.mark_as_completed("SHA256 sum validation error.");
+                LOG_ERROR << "File not valid: SHA256 sum doesn't match expectation "
+                          << m_tarball_path << "\nExpected: " << m_sha256
+                          << "\nActual: " << sha256sum << "\n";
+            }
+            return;
         }
-        else
+        if (!m_md5.empty())
```

Comment:
this is changing the behavior, previously the `MD5` was not checked if the `SHA256` was provided

Reply:
there is a return above